From 688c21e007cae5dceb2084e04d8131678cc6b71a Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Wed, 29 May 2019 14:07:05 +0800
Subject: [PATCH] DRVLIB_SW-2340 - Add gphy fw version printout

Level is set to info to make it visible for all users, as
this is useful for info and debugging purpose.
---
 drivers/net/phy/lantiq.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/phy/lantiq.c b/drivers/net/phy/lantiq.c
index c6fd680e5244..e263c6815c8b 100644
--- a/drivers/net/phy/lantiq.c
+++ b/drivers/net/phy/lantiq.c
@@ -24,6 +24,7 @@
 
 #define MII_VR9_11G_IMASK	0x19	/* interrupt mask */
 #define MII_VR9_11G_ISTAT	0x1a	/* interrupt status */
+#define MII_PHY_FWV		0x1e	/* firmware version */
 
 #define INT_VR9_11G_WOL		BIT(15)	/* Wake-On-LAN */
 #define INT_VR9_11G_ANE		BIT(11)	/* Auto-Neg error */
@@ -193,6 +194,9 @@ static int p31g_gphy_config_init(struct phy_device *phydev)
 	__set_bit(ETHTOOL_LINK_MODE_2500baseT_Full_BIT,
 		  phydev->extended_advertising);
 
+	dev_info(&phydev->mdio.dev, "fw version 0x%04X\n",
+		 phy_read(phydev, MII_PHY_FWV));
+
 	return 0;
 }
 
